python/ncnn/model_zoo/yolov8.py

```diff
@@ -0,0 +1,312 @@
+# Tencent is pleased to support the open source community by making ncnn available.
+#
+# Copyright (C) 2020 THL A29 Limited, a Tencent company. All rights reserved.
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import time
+import numpy as np
+import ncnn
+from .model_store import get_model_file
+from ..utils.objects import Detect_Object
+from ..utils.functional import *
+
+class YoloV8s:
+    def __init__(
+        self,
+        target_size=640,
+        prob_threshold=0.25,
+        nms_threshold=0.45,
+        num_threads=1,
+        use_gpu=False,
+    ):
+        self.target_size = target_size
+        self.prob_threshold = prob_threshold
+        self.nms_threshold = nms_threshold
+        self.num_threads = num_threads
+        self.use_gpu = use_gpu
+
+        self.reg_max = 16
+        self.mean_vals = []
+        self.norm_vals = [1 / 255.0, 1 / 255.0, 1 / 255.0]
+
+        self.net = ncnn.Net()
+        self.net.opt.use_vulkan_compute = self.use_gpu
+        self.net.opt.num_threads = self.num_threads
+
+        # original pretrained model from https://github.com/ultralytics/ultralytics
+        # the ncnn model https://github.com/nihui/ncnn-assets/tree/master/models
+        # self.net.load_param(get_model_file("yolov8s.param"))
+        # self.net.load_model(get_model_file("yolov8s.bin"))
+        self.net.load_param("/home/ubuntu/Build/ncnn/python/models/yolov8s.param")
+        self.net.load_model("/home/ubuntu/Build/ncnn/python/models/yolov8s.bin")
```

Comment:
do not hardcode path

Reply:
> 

How about now

Reply:
The link to yolov8s.param is not accessible. Could you please reupload it?